actionpack/test/dispatch/executor_test.rb

```diff
@@ -36,42 +36,18 @@ def test_returned_body_object_always_responds_to_close_even_if_called_twice
     body.close
   end
 
-  def test_returned_body_object_behaves_like_underlying_object
-    body = call_and_return_body do
-      b = MyBody.new
-      b << "hello"
-      b << "world"
-      [200, { "Content-Type" => "text/html" }, b]
-    end
-    assert_equal 2, body.size
-    assert_equal "hello", body[0]
-    assert_equal "world", body[1]
-    assert_equal "foo", body.foo
-    assert_equal "bar", body.bar
```

Comment:
`size`, `[]`, etc are not part of the SPEC for the body object. 

I could not find usages of these in code outside the tests, so I don't think other middlewares are relying on these methods: https://github.com/rails/rails/blob/3a8b9baa51125567053869375147b69f31874c8f/railties/test/commands/middleware_test.rb#L64-L91

Removing these can help ensure this won't be introduced in the future.

Reply:
:+1: I think this test (and the other removed one) are effectively just testing the implementation of BoxyProxy